code/game/objects/items/tools/flame_tools.dm

```diff
@@ -432,17 +432,89 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 //  WEED  //
 ////////////
 /obj/item/clothing/mask/cigarette/weed
-	name = "weed joint"
+	name = "\the weed joint"
+	icon_state = "blunt_off"
+	icon_on = "blunt_on"
+	icon_off = "blunt_off"
+	item_state = "blunt_on"
 	desc = "A rolled-up package of ambrosia vulgaris, aka space weed, in some smooth paper; you sure this is legal dude?"
 	chem_volume = 39
 	smoketime = 20 MINUTES
+	actions_types = list(/datum/action/item_action/hotbox)
+	var/puff_cooldown = 10 SECONDS
+	var/can_puff = TRUE
+	var/datum/effect_system/smoke_spread/smoke
 
 /obj/item/clothing/mask/cigarette/weed/Initialize()
 	. = ..()
 	reagents.add_reagent("space_drugs",15)
 	reagents.add_reagent("bicaridine", 8)
 	reagents.add_reagent("kelotane", 1)
 
+/obj/item/clothing/mask/cigarette/weed/verb/puff()
+	set name = "Hotbox"
+	set desc = "Puff a fat cloud"
+	set category = "Smoke"
+	set src in usr
+	var/mob/living/carbon/human/smoker = usr
+	var/datum/effect_system/smoke_spread/smoke = new /datum/effect_system/smoke_spread/weed
+	if(!usr || usr.is_mob_incapacitated(TRUE))
+		return
+	if(smoker.wear_mask != src)
+		return
+	if(!can_puff)
+		to_chat(smoker,SPAN_NOTICE("Your lungs lack fresh air, chill man."))
+		return
+
+	if(icon_on == "blunt_off")
```

Comment:
this is gross and you shouldn't be checking icon states to see if something is true/false